MFVNeutralino/plugins/Vertexer.cc

```diff
@@ -1,891 +1,2046 @@
+    max_track_vertex_sig(cfg.getParameter<double>("max_track_vertex_sig")),
+    min_track_vertex_sig_to_remove(cfg.getParameter<double>("min_track_vertex_sig_to_remove")),
+    remove_one_track_at_a_time(cfg.getParameter<bool>("remove_one_track_at_a_time")),
+    max_nm1_refit_dist3(cfg.getParameter<double>("max_nm1_refit_dist3")),
+    max_nm1_refit_distz(cfg.getParameter<double>("max_nm1_refit_distz")),
+    max_nm1_refit_count(cfg.getParameter<int>("max_nm1_refit_count")),
+    trackrefine_sigmacut(cfg.getParameter<double>("trackrefine_sigmacut")),
+    trackrefine_trimmax(cfg.getParameter<double>("trackrefine_trimmax")),
+    histos(cfg.getUntrackedParameter<bool>("histos", false)),
+    histos_noshare(cfg.getUntrackedParameter<bool>("histos_noshare", false)),
+    histos_output0(cfg.getUntrackedParameter<bool>("histos_output0", false)),
+    histos_output1(cfg.getUntrackedParameter<bool>("histos_output1", false)),
+    histos_output2(cfg.getUntrackedParameter<bool>("histos_output2", false)),
+    verbose(cfg.getUntrackedParameter<bool>("verbose", false)),
+    module_label(cfg.getParameter<std::string>("@module_label"))
+{
+  if (n_tracks_per_seed_vertex < 2 || n_tracks_per_seed_vertex > 5)
+    throw cms::Exception("MFVVertexer", "n_tracks_per_seed_vertex must be one of 2,3,4,5");
+
+  produces<reco::VertexCollection>();
+  produces<VertexerPairEffs>();
+  produces<reco::TrackCollection>("seed"); // JMTBAD remove me
+  produces<reco::TrackCollection>("inVertices");
+
+  if (histos) {
+    edm::Service<TFileService> fs;
+
+    h_n_seed_vertices                = fs->make<TH1F>("h_n_seed_vertices",                "",  50,   0,    200);
+    h_seed_vertex_track_weights      = fs->make<TH1F>("h_seed_vertex_track_weights",      "",  21,   0,      1.05);
+    h_seed_vertex_chi2               = fs->make<TH1F>("h_seed_vertex_chi2",               "",  20,   0, max_seed_vertex_chi2);
+    h_seed_vertex_ndof               = fs->make<TH1F>("h_seed_vertex_ndof",               "",  10,   0,     20);
+    h_seed_vertex_x                  = fs->make<TH1F>("h_seed_vertex_x",                  "", 20,  -1,      1);
+    h_seed_vertex_y                  = fs->make<TH1F>("h_seed_vertex_y",                  "", 20,  -1,      1);
+    h_seed_vertex_rho                = fs->make<TH1F>("h_seed_vertex_rho",                "", 20,   0,      2);
+    h_seed_vertex_phi                = fs->make<TH1F>("h_seed_vertex_phi",                "",  20,  -3.15,   3.15);
+    h_seed_vertex_z                  = fs->make<TH1F>("h_seed_vertex_z",                  "",  20, -20,     20);
+    h_seed_vertex_r                  = fs->make<TH1F>("h_seed_vertex_r",                  "", 20,   0,      2);
+    h_seed_vertex_paird2d            = fs->make<TH1F>("h_seed_vertex_paird2d",            "", 20,   0,      0.2);
+    h_seed_vertex_pairdphi           = fs->make<TH1F>("h_seed_vertex_pairdphi",           "", 20,  0,   3.14);

```

Comment:
I forget, did we change the binning on these because of memory issues? I think it would be useful to change the number of bins back for some of these (particularly `paird2d` and `pairdphi`), since they do get used elsewhere in the analysis. This would likely be true also of the noshare equivalents.

Reply:
If we do revert to the old binning for dphi, we should also remove the absolute value when we fill the histogram.

Reply:
I think I change it before I learned how to use Rebin(10) with comparehists.py. It might not have anything to do with memories? and for dphi, I change to an absolute one because we have only been calling dpi from an absolute one. 

Either way works for me. Let me know if you would like to keep the old/new one?

Reply:
Okay, I don't feel strongly about x,y,rho,phi,z,r. It's fine to use the coarser one for those.

For paird2d and pairdphi, let's adopt the finer granularity and use the non-abs value version, since we can always do rebinning as you say. We've use these hists for determining the overlap efficiency in our background template construction, so it seems safer to have the finer bins for these.

Reply:
Okay, I will revert the paird2d and pairdphi binning.

Reply:
Done in https://github.com/DisplacedVertices/cmssw-usercode/commit/c23e5fc6ca66d0751292d8d56c1f5b9cda371a72, right?

I'll mark as resolved, but reopen if I am wrong.

Reply:
Done.